update-gke-deployment/action.yaml

```diff
@@ -51,3 +51,9 @@ runs:
       shell: bash
       run: |-
         kubectl rollout status deployment "${{ inputs.deployment-name }}-${{ inputs.deployment-environment }}" --timeout="${{ inputs.deployment-rollout-timeout }}" --namespace="${{ inputs.deployment-namespace }}"
+    - name: Deployment failure
+      shell: bash
+      if: ${{ inputs.deployment-rollout-timeout }} == '10m'
```

Comment:
I'm not sure this `if` check will do what you are looking for. We only want to print this if the status of the step before `kubectl rollout status deployment` returned with a failure. The check you have here will print a failure if the `deployment-rollout-timeout` is set to `10m` which is just how long the previous step should wait until failure. I also think you'll run into an issue due to the previous step failing which means this following step will never execute as the entire workflow will exit before reaching this. One thing to check out would be how to test if a bash command failed and perform a different action if it did. One thing to keep in mind with saving previous command exit codes is that will capture the error and not instantly fail. This is desirable so you can print your log message but you should still exit the overall step with the same exit code returned from the `kubectl rollout status deployment` command.